kolibri/core/content/views.py

```diff
@@ -99,7 +133,7 @@ def get(self, request, zipped_filename, embedded_filepath):
 
         # restrict CSP to only allow resources to be loaded from the Kolibri host, to prevent info leakage
         # (e.g. via passing user info out as GET parameters to an attacker's server), or inadvertent data usage
-        host = request.build_absolute_uri(OPTIONS['Deployment']['URL_PATH_PREFIX']).strip("/")
+        host = get_host(request, parsed_referrer_url)
```

Comment:
Realized that actually this may not help in the only case that matters (loading an HTML page inside an iframe), as the referer header may not be set.

Reply:
Oh, I am incorrect - marvellous. Looks like the `Referer` header *is* set on the request for an iframe's src.